### ISSUE TITLE
Training Performance and Stability

### ISSUE THREAD

**Issue by lukemelas** [valeoai/ADVENT#15]
Hello, 

Thank you for your work on this repo! I have a quick question. When I train a model (either MinEnt or Advent), I find that the validation performance varies widely (sometimes 5-10 mIOU points) from snapshot to snapshot (where snapshots are taken every 2000 iterations with the default learning rate). Do you recall experiencing this type of variation in mIOU across snapshots?

If so, did you just report the score of the best single snapshot on the 500-image val set (i.e. take the best evaluation under the 'best' config for `cfg.TEST.MODE`)? 

It is possible that the performance stabilizes after more steps, but I am currently at iteration 60,000, so that seems unlikely at this point. Thank you! 

Note: I have read all the previous issues and I am not concerned with attaining 43.3 vs 43.8 or that sort of thing. I am concerned with performance varying more widely, from say 41 to 37 to 42 within a few thousand iterations.

**Comment by manrai** [comment]
Thank you for releasing this code! I also had the same question as @lukemelas.

**Comment by tuanhungvu** [comment]
Hello @lukemelas and @manrai , sorry for the late reply. Thank you for using our code.
Yes, we reported score of the best snapshot in the paper.
Regarding the varying performance, we did observe the same behavior. You can try training more (~250k iter), but I doubt that the target performance will degrade eventually. Normally our best snapshot is between 60-90k.
Best,
T-H

Best,
T-H

**Comment by lukemelas** [comment]
Thank you for your response! I will add to this thread if I have any more questions. For now, I am closing the issue.
